### DISCUSSION TITLE
Scaled font size calculation in page.get_texttrace()

### DISCUSSION THREAD

**Discussion by sky884** [pymupdf/PyMuPDF#2645] [Looking for help] [answered]
I'm using texttrace to extract individual characters, their formatting, origin points and bounding boxes. This has been working well, but I've come across a problem with a particular PDF. Texttrace shows a size value for the text in this PDF of 17.33, but Acrobat displays the text at 12.99. 

Inspecting the text with PDFXplorer shows a size value of 17.33 matching texttrace, but also a scaling transformation of 0.75 (actually the CTM shows 0.75 0 0 -0.75). This perhaps explains the difference between 17.33 and 12.99 as 17.33 * 0.75 = 12.99.

Extracting text from the same PDF with get_text("rawdict") gives a size value of 12.99.

Is there a way using PyMuPDF that I can extract the CTM value applied to this text, and so recalculate 17.33 as 12.99? Or some other method of getting to the 12.99 value from the 17.33 texttrace returns? I would prefer to use texttrace rather than get_text("rawdict") as it's faster and it gives a spacewidth value which might help me calculate character spacing.

PyMuPDF is excellent, many thanks for developing such a great product

**Comment by JorjMcKie** [comment]
Thank you for your post and your nice appreciation of PyMuPDF! Inspiring!

Let me check what is happening there. My goal is to return the same font size value in `get_texttrace()` as in the `get_text()` method (which is a direct result of MuPDF logic).
Inside `get_texttrace()` all information is available - including page CTM and text TRM. There unfortunately is now way to hand these matrices to the caller, so I am bound to do it correctly inside the method ... 😎.

**Reply by sky884** [reply]
Thanks for taking the time to look into this for me.

**Reply by JorjMcKie** [reply]
My code still looks harmless to me - can you please share that PDF example?

**Comment by sky884** [comment]
[texttrace example.pdf](https://github.com/pymupdf/PyMuPDF/files/12728325/texttrace.example.pdf)
Sure here's the file.

**Reply by JorjMcKie** [reply]
I have found the place where this happens - of course inside C-code.
I am preparing a fix for a version in near future.

**Reply by sky884** [reply, marked as answer]
Great, thanks for your help.
